src/tests/managers/test_config_manager.py

```diff
@@ -0,0 +1,73 @@
+from pathlib import Path
+from unittest.mock import patch, mock_open
+import unittest
+
+from cli.exceptions import MalformedJsonFile
+from cli.managers.config_manager import ConfigManager
+
+class TestConfigManager(unittest.TestCase):
+
+    def test_get_applications(self):
+        config_manager = ConfigManager(config_dir=Path('dummy'))
+
+        dummy_apps_config = "{" \
+                                "\"app_1\" : {\"path\": \"/some/path/\"," \
+                                              "\"application_id\": 1}," \
+                                "\"app_2\" : {\"path\": \"/some/path/\"," \
+                                              "\"application_id\": 2}" \
+                            "}"
+
+        with patch('cli.managers.config_manager.open', mock_open(read_data=dummy_apps_config)),\
+             patch('pathlib.Path.is_file',return_value=True):
+            apps = config_manager.get_applications()
+            self.assertEqual(len(apps), 2)
+            for app in apps:
+                self.assertIn('name', app)
+
+    def test_get_applications_no_config(self):
+        config_manager = ConfigManager(config_dir=Path('dummy'))
+
+        with patch('cli.managers.config_manager.open', mock_open(read_data="{}")),\
+             patch('pathlib.Path.is_file',return_value=False):
+            apps = config_manager.get_applications()
+            self.assertEqual(len(apps), 0)
+
+    def test_get_applications_invalid_config(self):
+        config_manager = ConfigManager(config_dir=Path('dummy'))
+        malformed_apps_config = "{" \
+                            "\"app_1\"  {\"path\" \"/some/path/\"," \
+                            "\"application_id\" 1}," \
+                            "\"app_2\"  {\"path\" \"/some/path/\"," \
+                            "\"application_id\" 2}" \
+                            "}"
+
+        with patch('cli.managers.config_manager.open', mock_open(read_data=malformed_apps_config)),\
+             patch('pathlib.Path.is_file',return_value=True):
+
+            self.assertRaises(MalformedJsonFile, config_manager.get_applications)
+
+    def test_add_application(self):
+        config_manager = ConfigManager(config_dir=Path('dummy'))
```

Comment:
What has this to do with the function you added, same for some more tests.

Reply:
Fixed